### PR TITLE
[13.0] [MIG] barcodes_generator_product: Migration to v13.0

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT26762

please @pedrobaeza @joao-p-marques review this

### CODE REVIEW

**Review by legalsylvain — CHANGES_REQUESTED** [review]
Hi.
Why do you remove demo file ? 
It allows users to test more easely on local test database / runbot / etc..

Could you restore them ? thanks.

File: barcodes_generator_product/models/product_template.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+        # this is needed to set given values to first variant after creation
+        # these fields should be moved to product as lead to confusion
+        # (Ref. product module feature in Odoo Core)
+        related_vals = {}
+        for field in ["barcode_rule_id", "barcode_base"]:
+            if vals.get(field, False):
+                related_vals[field] = vals[field]
+            if related_vals:
+                template.write(related_vals)
```
Comment (rousseldenis) [inline barcodes_generator_product/models/product_template.py]: I'm quite perplex on this (a write in a create)

Comment (pedrobaeza) [reply barcodes_generator_product/models/product_template.py]: Please read code comment below these lines

Comment (pedrobaeza) [reply barcodes_generator_product/models/product_template.py]: @CarlosRoca13 what should be done is to deindent these 2 lines, as it shouldn't be executed only once.

**Comment by CarlosRoca13** [comment]
@legalsylvain recovered and adapted to v13 creation and make the change that @pedrobaeza said

File: barcodes_generator_product/README.rst [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+To use this module, you need to:
+
+* Go to a Product form (or a template form):
+
+1 for manual generation
+    * Set a Barcode Rule
+    * Set a Barcode Base
+    * click on the button 'Generate Barcode (Using Barcode Rule)'
+
+.. image:: https://raw.githubusercontent.com/barcodes_generator/static/description/product_template_manual_generation.png
```
Comment (joao-p-marques) [inline barcodes_generator_product/README.rst]: This link is not being generated correctly, but maybe that is a bug in `oca-gen-addon-readme`. The correct one should be:
```suggestion
.. image:: https://raw.githubusercontent.com/OCA/stock-logistics-barcode/13.0/barcodes_generator_product/static/description/product_template_manual_generation.png
```

**Review by joao-p-marques — COMMENTED** [review]
Code LGTM :+1: 

However, I can't test on runbot. I think the demo data is not being loaded correctly. Can you test that? 
If not, maybe I am doing something wrong. Could you review the documentation to check that everything is correct as for v13?

File: barcodes_generator_product/README.rst (hunk 2 above)
Comment (CarlosRoca13) [reply barcodes_generator_product/README.rst]: Yes sorry, it was bad defined on USAGE.rst solved

Comment (pedrobaeza) [reply barcodes_generator_product/README.rst]: You must use the relative link in fragments (`../static/src/<etc>`), and let the script to generate the absolute one.

**Comment by pedrobaeza** [comment]
Is everything OK now? Can this be rechecked?

**Comment by joao-p-marques** [comment]
@CarlosRoca13 I still can't test on the runbot. The `barcode.nomenclature` and respective `barcode.rule` are being correctly created:
image
However, I can't se the fields in `product.template` or `product.product`:
image

**Comment by CarlosRoca13** [comment]
@joao-p-marques sorry I should explain on configure that you need to add the user on "Generate Barcodes" group. Now is added

**Review by joao-p-marques — APPROVED** [review]
Ok @CarlosRoca13 
Now tested functionally :+1:

**Comment by CarlosRoca13** [comment]
Please @rousseldenis @legalsylvain take a look for finish with this PR

File: barcodes_generator_product/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+# Copyright (C) 2018 Komit (https://komit-consulting.com)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Generate Barcodes for Products",
+    "summary": "Generate Barcodes for Products (Templates and Variants)",
+    "version": "13.0.1.0.0",
+    "category": "Tools",
+    "author": "GRAP," "La Louve," "Odoo Community Association (OCA)",
```
Comment (victoralmau) [inline barcodes_generator_product/__manifest__.py]:
```suggestion
    "author": "GRAP, La Louve, Odoo Community Association (OCA)",
```

File: barcodes_generator_product/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    "website": "https://www.odoo-community.org",
```
Comment (victoralmau) [inline barcodes_generator_product/__manifest__.py]:
```suggestion
    "website": "https://github.com/OCA/stock-logistics-barcode",
```

File: barcodes_generator_product/i18n/es_ES.po [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+#. module: barcodes_generator_product
+#: model:product.product,weight_uom_name:barcodes_generator_product.product_product_variant_1
+#: model:product.product,weight_uom_name:barcodes_generator_product.product_product_variant_2
+#: model:product.template,weight_uom_name:barcodes_generator_product.product_product_variant_1_product_template
+#: model:product.template,weight_uom_name:barcodes_generator_product.product_template_mono_variant
+msgid "kg"
+msgstr ""
+
+#~ msgid "barcode.rule"
+#~ msgstr "regla.barra de codigos"
```
Comment (victoralmau) [inline barcodes_generator_product/i18n/es_ES.po]:
```suggestion
```

File: barcodes_generator_product/i18n/fr.po [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+msgid "kg"
+msgstr ""
+
+#, fuzzy
+#~| msgid "Template with Generated Barcode (Mono Variant)"
+#~ msgid "Product with Generated Barcode (Variant 2 - Black)"
+#~ msgstr "Article avec Code Barre Généré (Mono Variante)"
+
+#~ msgid "barcode.rule"
+#~ msgstr "barcode.rule"
```
Comment (victoralmau) [inline barcodes_generator_product/i18n/fr.po]:
```suggestion
```

File: barcodes_generator_product/i18n/nl_NL.po [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+#. module: barcodes_generator_product
+#: model:product.product,weight_uom_name:barcodes_generator_product.product_product_variant_1
+#: model:product.product,weight_uom_name:barcodes_generator_product.product_product_variant_2
+#: model:product.template,weight_uom_name:barcodes_generator_product.product_product_variant_1_product_template
+#: model:product.template,weight_uom_name:barcodes_generator_product.product_template_mono_variant
+msgid "kg"
+msgstr ""
+
+#~ msgid "barcode.rule"
+#~ msgstr "barcode.rule"
```
Comment (victoralmau) [inline barcodes_generator_product/i18n/nl_NL.po]:
```suggestion
```

**Review by victoralmau — CHANGES_REQUESTED** [review]

**Comment by CarlosRoca13** [comment]
@victoralmau changes done

**Review by victoralmau — APPROVED** [review]

**Pull request merged**